src/basic_recipes/bracket.jl

```diff
@@ -0,0 +1,146 @@
+"""
+    bracket(x1, y1, x2, y2; kwargs...)
+    bracket(x1s, y1s, x2s, y2s; kwargs...)
+    bracket(point1, point2; kwargs...)
+    bracket(vec_of_point_tuples; kwargs...)
+
+Draws a bracket between each pair of points (x1, y1) and (x2, y2) with a text label at the midpoint.
+
+By default each label is rotated parallel to the line between the bracket points.
+
```

Comment:
```suggestion

In order to implement a new style for brackets, overload `Makie.bracket_bezierpath(::Val{:mybracketstyle}, p1, p2, d, width)`, 
which should return a [`BezierPath`](@ref).
```

Reply:
I don't want to make that API, yet, in case I find there's more to it than I currently have.